doc/building.md

```diff
@@ -1503,6 +1503,68 @@ If you need general help or advice about developing for the JDK, you can also
 contact the Adoption Group. See the section on [Contributing to OpenJDK](
 #contributing-to-openjdk) for more information.
 
+## Reproducible Builds
+
+Build reproducibility is the property of getting exactly the same bits out when
+building, every time, independent on who builds the product, or where. This is
+for many reasons a harder goal than it initially appears, but it is an important
+goal, for security reasons and others. Please see [Reproducible Builds](
+https://reproducible-builds.org) for more information about the background and
+reasons for reproducible builds.
+
+Currently, it is not possible to build OpenJDK fully reproducibly, but getting
+there is an ongoing effort. There are some things you can do to minimize
+non-determinism and make a larger part of the build reproducible:
+
+  * Turn on build system support for reproducible builds
+
+Add the flag `--enable-reproducible-builds` to your `configure` command line.
+This will turn on support for reproducible builds where it could otherwise be
+lacking.
+
+  * Do not rely on `configure`'s default adhoc version strings
+
+Default adhoc version strings OPT segment include user name, source directory
+and timestamp. You can either override just the OPT segment using
+`--with-version-opt=<any fixed string>`, or you can specify the entire version
+string using `--with-version-string=<your version>`.
+
+  * Specify how the build sets `SOURCE_DATE_EPOCH`
+
+The JDK build system will set the `SOURCE_DATE_EPOCH` environment variable
+during building, depending on the value of the `--with-source-date` option for
+`configure`. The default value is `updated`, which means that
+`SOURCE_DATE_EPOCH` will be set to the current time each time you are running
+`make`.
+
+The [`SOURCE_DATE_EPOCH` environment variable](
+https://reproducible-builds.org/docs/source-date-epoch/) is an industry
+standard, that many tools, such as gcc, recognize, and use in place of the
+current time when generating output.
+
+For reproducible builds, you need to set this to a fixed value. You can use the
+special value `version` which will use the nominal release date for the current
+JDK version, or a value describing a date, either an epoch based timestamp as an
+integer, or a valid ISO-8601 date.
+
+**Hint:** If your build environment already sets `SOURCE_DATE_EPOCH`, you can
+propagate this using `--with-source-date=$(SOURCE_DATE_EPOCH)`.
```

Comment:
Shouldn't this example be $SOURCE_DATE_EPOCH, or at least ${SOURCE_DATE_EPOCH} as this would most likely be run in a shell? Or are you trying to send in the make variable reference verbatim to be evaluated in the makefiles?

Reply:
Yeah, no, you're right. It's environmental damage due to working too long in makefiles. :) I'll fix, thanks for catching it.